### PR TITLE
ZER-54 Add validation and authorization to a login page

### PR DESCRIPTION
**Pull request by aboriiisova**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/controllers/user/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    # GET /resource/sign_in
+
+    # POST /resource/sign_in
+    def create
+      user = User.find_by_email(params[:email])
+      if user&.authenticate(params[:password])
+        session[:user_id] = user.id
+        redirect_to root_url
+      else
+        flash.now.alert = 'Email or password is invalid'
```
Comment (maksymst) [inline app/controllers/user/sessions_controller.rb:16]: Localize it

File: app/controllers/user/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    before_action :configure_sign_in_params, only: [:create]
+
+    # GET /resource/sign_in
+
+    # POST /resource/sign_in
+    def create
+      user = User.find_by_email(params[:email])
+      if user&.authenticate(params[:password])
+        session[:user_id] = user.id
+        redirect_to root_url
```
Comment (maksymst) [inline app/controllers/user/sessions_controller.rb:14]: It should be [this](https://github.com/ita-social-projects/ZeroWaste/blob/master/app/controllers/application_controller.rb#L18)

File: app/controllers/user/confirmations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
```
Comment (maksymst) [inline app/controllers/user/confirmations_controller.rb:1]: remove controllers that are not in use

File: db/migrate/20210506182617_add_lockable_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class AddLockableToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :failed_attempts, :integer, default: 0
+    add_column :users, :unlock_token, :string
+    add_column :users, :locked_at, :datetime
```
Comment (maksymst) [inline db/migrate/20210506182617_add_lockable_to_users.rb:5]: use `change _table`

**Review by maksymst — CHANGES_REQUESTED** [review]

File: db/migrate/20210506182617_add_lockable_to_users.rb (hunk 4 above)
Comment (aboriiisova) [reply db/migrate/20210506182617_add_lockable_to_users.rb]: Fixed

File: app/controllers/user/confirmations_controller.rb (hunk 3 above)
Comment (aboriiisova) [reply app/controllers/user/confirmations_controller.rb]: Removed

File: app/controllers/user/sessions_controller.rb (hunk 2 above)
Comment (aboriiisova) [reply app/controllers/user/sessions_controller.rb]: Changed

File: app/controllers/user/sessions_controller.rb (hunk 1 above)
Comment (aboriiisova) [reply app/controllers/user/sessions_controller.rb]: Fixed

**Review by maksymst — DISMISSED** [review]

**Review by haliapats — APPROVED** [review]

**Review by maksymst — APPROVED** [review]

**Pull request merged**